### PR TITLE
Feature/sign-up

### PR DESCRIPTION
**Pull request by gabrielgomesml**
## What I did

- Create Sign Up flow using redux.

## How to test

- First of all make sure you are running the server in your machine. You can check the steps here: https://github.com/gabrielgomesml/training_project_server/pull/1
- Open the project and use the command `nvm use`
- Run a `yarn` or `yarn install`
- After that run the command `yarn start` to start the project
- Click in "Cadastre-se" and test the flow of creating a new user.

P.S: I didn't connect the success and failure action in the request, i intend to do this later. Also, no photo adding for now.

### CODE REVIEW

File: src/pages/SignUp/index.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+} from './style';
+import { Input } from './Components/Input';
+import { Button } from '../../components';
+import { State } from '../../store/reducers';
+
+export const SignUp: React.FC = () => {
+  const dispatch = useDispatch();
+  const [photoPreview, setPhotoPreview] = useState('');
+
+  const { toggleStep, updateValue, loadRequest } = bindActionCreators(
```
Comment (JonasMazza) [inline src/pages/SignUp/index.tsx:28]: There's nothing wrong with this pattern, but my suggestion would be to create an `useSignupActions` hook that returns the binded actions.

File: src/store/types/signUpTypes.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+export type Action =
+  | ToggleAction
+  | UpdateAction
+  | RequestAction
+  | SucessAction
+  | FailureAction;
+
+export type postActionCreator = (
```
Comment (JonasMazza) [inline src/store/types/signUpTypes.ts:48]: Should be capitalized

File: src/store/actions-creators/index.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+export const updateValue = (label: string, value: string | File) => ({
+  type: SignUpTypes.UPDATE_VALUE,
+  payload: {
+    label,
+    value,
+  },
+});
+
+export const loadRequest: postActionCreator =
+  (data: RequestType) => async () => {
```
Comment (JonasMazza) [inline src/store/actions-creators/index.ts:22]: Check this, if you haven't yet:  https://redux.js.org/tutorials/fundamentals/part-6-async-logic
And this (Dan Abramanov's answer): https://stackoverflow.com/questions/34570758/why-do-we-need-middleware-for-async-flow-in-redux/34599594#34599594

**Review by JonasMazza — APPROVED** [review]

File: src/pages/SignUp/index.tsx (hunk 1 above)
Comment (gabrielgomesml) [reply src/pages/SignUp/index.tsx:28]: I will merge this PR and make the requests solutions in another branch.

**Pull request merged**